blog/2020-09-21-setup-bitcoin-cloud-node.md

```diff
@@ -0,0 +1,289 @@
+If you don't have cloud platform experience at all you might have to consult the [Google Cloud documentation](https://cloud.google.com/compute/docs) on the way.
+
+### Setup Recommendations
+
+Instance recommendations: 
+
+- Use a `e2-standard-4 (4 vCPUs, 16 GB memory)` instance for installation and initial sync. **This machine will only be used for the first 12-24 hours with costs less than 4 AUD in total.**
+- Use a `e2-small` instance (at least 2GB RAM) for running the node AFTER the initial sync. This costs about 33 AUD per month given 24/7 uptime.
+
+Why do we recommend this?
+
+We want to be economical, but we cannot wait forever. Syncing with a node < 8GB RAM takes significant time, we are talking weeks here not days. Running the strong machine for 12ish hours is actually cheaper than running a small machine for 1.5+ weeks.
+For more details on machine costs feel free to play with Google's [pricing calculator](https://cloud.google.com/products/calculator).
+
+More tutorial specs & quirks:
+
+- We use `Debian` for our machine. Feel free to use a different distribution, but be aware that you might have to adapt tutorial-steps on the way.
+- We use an `additional HDD` that we attach to the instance instead of upping the instance's SSD because it is slightly cheaper.
+
+Notes on costs:
+
+Total cost point is about 33 AUD per month for the e2-small instance + 500 GB HDD when running 24/7.
+To give this a try for a month you should calculate with about 40 AUD of costs, assuming that you only run the e2-standard machine for the first 12-24h and then run an e2 machine.
+We feel these costs are not really a big deal for the sake of trying something awesome, but you should be aware that there are costs. 
+If you want to go for a cheaper option you can sync on e.g. an external disk at home or on a Raspy.
+
+Instead of doing the setup yourself you can opt for using a pre-configured, pre-synced image provided on [Google Cloud Marketplace](https://console.cloud.google.com/marketplace/details/techlatest-public/bitcoin-fullnode).
+We opt for doing the setup ourselves to be in full control of what is going on.
+
+### Initial Google Instance Setup
+
+We refer to Bitcoin Core as bitcoind, as in bitcoin daemon, throughout this tutorial.
+
+#### Instance setup
+
+1. Create a `e2-standard-4 (4 vCPUs, 16 GB memory)` instance, provision it with `Debian`.
+2. [Add a 500 GB harddisk to the instance](https://cloud.google.com/compute/docs/disks/add-persistent-disk#create_disk) (you can do this during instance setup at the bottom or later).
+3. [Configure a static external IP address for the instance](https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address#assign_new_instance).
+
+#### Configure the Google Cloud firewall
+
+ℹ️ This tutorial assumes that you want to expose the RPC interface to the outside world, if you don't want to do this you can skip this step.
```

Comment:
Doesn't render well for me :(

![image](https://user-images.githubusercontent.com/5486389/94100269-86d32c00-fe70-11ea-9a50-36c316867893.png)

```suggestion
This tutorial assumes that you want to expose the RPC interface to the outside world, if you don't want to do this you can skip this step.
```

Reply:
Also, the bitcoind documentation strongly recommends to never expose the RPC interface to the outside world. Not sure if we want to talk about this here :grimacing: 

https://github.com/bitcoin/bitcoin/blob/master/doc/JSON-RPC-interface.md#security

> Securing remote network access: You may optionally allow other computers to remotely control Bitcoin Core by setting the rpcallowip and rpcbind configuration parameters. These settings are only meant for enabling connections over secure private networks or connections that have been otherwise secured (e.g. using a VPN or port forwarding with SSH or stunnel). **Do not enable RPC connections over the public Internet.** Although Bitcoin Core's RPC interface does use authentication, it does not use encryption, so your login credentials are sent as clear text that can be read by anyone on your network path. Additionally, the RPC interface has not been hardened to withstand arbitrary Internet traffic, so changing the above settings to expose it to the Internet (even using something like a Tor onion service) could expose you to unconsidered vulnerabilities. See bitcoind -help for more information about these settings and other settings described in this document.

Reply:
I changed it to only expose `8333` and removed the section with `rpc` username and password altogether. 
When writing the blogpost on how to use nectar we could add a section with `ssh` port forwarding to communicate with a potential cloud node (or add an additional blogpost for that).